packages/KModalFullscreen/KModalFullscreen.vue

```diff
@@ -0,0 +1,268 @@
+<template>
+  <div
+    v-if="isVisible"
+    :aria-label="title"
+    class="k-modal isOpen"
+    role="dialog"
+    aria-modal="true">
+    <div
+      class="k-modal-dialog"
+    >
+      <div class="k-modal-header">
+        <div
+          v-if="$scopedSlots.title || !hideTitle"
+          class="k-modal-header-description mb-5"
+          role="heading"
+          aria-level="2">
+          <div class="k-modal-title">
+            <span class="header-icon">
+              <slot name="header-icon"/>
+            </span>
+            <span class="header-content">
+              <slot name="header-content">{{ title }}</slot>
+            </span>
+          </div>
+          <div
+            class="k-modal-action ml-3">
+            <KButton
+              :appearance="cancelButtonAppearance"
+              @click="close"
+              @keyup.esc="close">
```

Comment:
With this component mainly being used as a form (at least in Konnect) should we remove the `escape` keypress? My thought is that if you're filling out a form and accidentally press escape you're going to lose all the form data you already filled out.

cc: @kaiarrowood 

Additionally, we could add a [window `onBeforeUnload`](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload) event when the modal is opened (and unbind on modal close and proceed) that will prompt the user if they try closing the window

Reply:
@adamdehaven: haven't removed `escape` keypress yet, please let me know if its confirmed to remove it?

Reply:
@kaiarrowood my thought here is since this will always be used as a form in Konnect, we should make it deliberate to close the modal (e.g. unbind the escape and enter keys) and rely on tab for keyboard nav.

If we leave escape in place, you could be filling out the form and hit escape on accident and lose all of your changes.

If we leave enter in place, the form inside the slot has no knowledge of the proceed button, I think, unless it's slotted.

Reply:
I don't think I agree with you here. Users expect that when they are on a form they can tab through and enter the values for fields they need and then hit ENTER to submit the form without being required to tab through the rest of the form fields to navigate back around to the buttons (which are stupidly at the top). We can only stupid-proof things so far. If a user accidentally clicks the Cancel/Ok buttons their content will be lost too. We should assume that when a user pushes / clicks a button, it is intentional.

Reply:
Lastly, if the `keyup` event is bound to the modal rather than the buttons, we still correctly `emit` regardless of whether or not the buttons are clicked. Remember that all we are doing is emitting events when buttons are clicked, the consumer app is responsible for determining what happens on those actions.

Reply:
Also, while trying keyboard nav I found that after tabbing to the bottom of the form, tabbing one more time actually took me to the content underneather modal rather than back to the top of the modal. No idea if that can be fixed.

Reply:
You could do something like add `pointer-events: none` to the body, then `pointer-events: auto` on the modal but only while it's open. Always set back on close.

Reply:
Can we bind this event to the main div instead of the button?